src/lib.rs

```diff
@@ -238,83 +265,13 @@ where
     }
 }
 
-/// Range counting down
-impl<Idx> Range<Idx, Downwards>
-where
-    Idx: Clone + PartialEq + One + AddAssign + SubAssign,
-{
-    /// Collect range into a container
-    ///
-    /// Works for any container type that implements [`FromIterator`].
-    ///
-    /// # Examples
-    ///
-    /// Basic usage, creating a [`Vec`]. See also [`Range::to_vec`]
-    ///
-    /// ```rust
-    /// extern crate a_range;
-    ///
-    /// let collection: Vec<_> = a_range::from(42).down_to(38).collect();
-    ///
-    /// assert_eq!(collection, vec![42, 41, 40, 39, 38]);
-    /// ```
-    ///
-    /// Using `collect()` to make a [`std::collections::HashSet`]
-    ///
-    /// ```rust
-    /// extern crate a_range;
-    /// use std::collections::HashSet;
-    ///
-    /// let collection: HashSet<_> = a_range::from(42).down_to(38).collect();
-    ///
-    /// assert!(collection.contains(&39));
-    /// ```
-    pub fn collect<B>(self) -> B
```

Comment:
I think we lost tests for collect/to_vec on downward ranges? Can you paste them in as a second example in the new generic impl?